nemo_aligner/utils/distributed.py

```diff
@@ -82,6 +86,53 @@ def broadcast_2d_tensor(tensor, src, group, dtype=torch.float32):
     return tensor
 
 
+_DTYPE_TO_ENUM = {
+    torch.float32: 0,
+    torch.float16: 1,
+    torch.bfloat16: 2,
+    torch.int32: 3,
+    torch.int64: 4,
+}
+
+_ENUM_TO_DTYPE = {enum: dtype for dtype, enum in _DTYPE_TO_ENUM.items()}
+
+
+def broadcast_tensor(tensor, src, group, dtype: torch.dtype | None = None):
+    f"""
+    Broadcast any tensor from the src rank to every other rank in the given group.
+    All the ranks that send or receive data must call this function.
+    
+    dtype: Optional dtype, {dir(torch)}
+    """
+    if torch.distributed.get_rank() == src:
+        tensor = tensor.cuda()
+        if dtype:
+            tensor = tensor.to(dtype)
+
+        input_dtype_enum = torch.tensor(_DTYPE_TO_ENUM.get(tensor.dtype, 0), device="cuda", dtype=torch.int32)
```

Comment:
is there a way to get dtype to int automatically using torch?

Reply:
not that i'm aware of. if you know some util let me know.

the only thing i found in pytorch src itself was [this](https://github.com/pytorch/pytorch/blob/46f158bfbc5c3d6020f8458ae749b89addc7ffa8/torch/_export/converter.py#L194), but it's protected, so I don't know if we want to depend on a non-public API